src/transactions/feed/JumpstartTransactionDetailsScreen.tsx

```diff
@@ -104,47 +137,88 @@ function JumpstartTransactionDetailsScreen({ route }: Props) {
       if (!isDeposit) {
         return
       }
-      return await getClaimDataAndStatus(
+
+      const { beneficiary, index, claimed } = await getClaimDataAndStatus(
         jumpstartContractAddress as Address,
-        transaction.networkId,
+        networkId,
         transaction.transactionHash as Hash
       )
+
+      if (claimed) {
+        return { claimed: true }
+      }
+
+      const reclaimTx: TransactionRequest = {
+        from: walletAddress as Address,
+        to: jumpstartContractAddress as Address,
+        value: BigInt(0),
+        data: encodeFunctionData({
+          abi: walletJumpstart.abi,
+          functionName: 'reclaimERC20',
+          args: [beneficiary, BigInt(index)],
+        }),
+      }
+
+      const preparedTransactions = await prepareTransactions({
+        feeCurrencies,
+        baseTransactions: [reclaimTx],
+      })
+
+      const resultType = preparedTransactions.type
+      switch (resultType) {
+        case 'need-decrease-spend-amount-for-gas': // fallthrough on purpose
+        case 'not-enough-balance-for-gas':
+          return {
+            claimed: false,
+            error: true,
```

Comment:
instead of returning an error in a success state, can we throw an error so that this goes through the generic error handling logic?

Reply:
Yeah, I didn't like that I'm missing the `claimed` information. I wanted to differentiate this error case when the error happens before and we don't know the claim status.
But at the end, I'm handling those scenarios as the same. So you are right, I'm gonna throw an error here.